pkg/yurthub/kubernetes/serializer/serializer.go

```diff
@@ -39,27 +43,211 @@ import (
 	"k8s.io/klog"
 )
 
+var (
+	cacheCRDRESTMapperKey = "_internal/cache-manager/cache-crd-restmapper.conf"
+	sepForGVR             = "/"
+)
+
 // YurtHubSerializer is a global serializer manager for yurthub
 var YurtHubSerializer = NewSerializerManager()
 
-// UnsafeDefaultRESTMapper is only used to check whether the GVK is in the scheme according to the GVR information
-var UnsafeDefaultRESTMapper = NewDefaultRESTMapperFromScheme()
+// YurtHubRESTMapperManager is a global RESTMapper for yurthub
+var YurtHubRESTMapperManager = NewRESTMapperManager()
+
+// DynamicRESTMapper is a mapper that can dynamically update the mapping relationship between GVK and GVR
+type DynamicRESTMapper struct {
+	sync.Mutex
+	// storageEnabled is used to determine whether to enable hard cache
+	storageEnabled bool
+	// Used to hard cache the resourceToKind
+	storage        storage.Store
+	resourceToKind map[schema.GroupVersionResource]schema.GroupVersionKind
+}
+
+func NewDynamicRESTMapper() *DynamicRESTMapper {
+	m := make(map[schema.GroupVersionResource]schema.GroupVersionKind)
+	s, err := factory.CreateStorage()
+	if err != nil {
+		klog.Errorf("could not create storage manager, %v", err)
+		return &DynamicRESTMapper{
+			storageEnabled: false,
+			storage:        nil,
+			resourceToKind: m,
+		}
+	}
+	// Recover the mapping relationship between GVR and GVK from the hard disk
+	b, err := s.Get(cacheCRDRESTMapperKey)
+
+	if err == nil && len(b) != 0 {
+		cacheMapper := make(map[string]string)
+		err = normaljson.Unmarshal(b, &cacheMapper)
+		if err != nil {
+			klog.Errorf("failed to get cached CRDRESTMapper, %v", err)
+		}
+
+		for gvrString, gvkString := range cacheMapper {
+			localInfo := strings.Split(gvrString, sepForGVR)
+			if len(localInfo) != 3 {
+				klog.Errorf("This %s is not standardized", gvrString)
+				continue
+			}
+			gvr := schema.GroupVersionResource{
+				Group:    localInfo[0],
+				Version:  localInfo[1],
+				Resource: localInfo[2],
+			}
+			gvk := schema.GroupVersionKind{
+				Group:   localInfo[0],
+				Version: localInfo[1],
+				Kind:    gvkString,
+			}
+			m[gvr] = gvk
+		}
+	}
+
+	return &DynamicRESTMapper{
+		storage:        s,
+		storageEnabled: true,
+		resourceToKind: m,
+	}
+}
+
+// Used to add and update the mapping relationship between GVR and GVK
+func (dm *DynamicRESTMapper) UpdateGVK(gvk schema.GroupVersionKind) error {
+	dm.Lock()
+	defer dm.Unlock()
+	kindName := strings.TrimSuffix(gvk.Kind, "List")
+	plural, singular := meta.UnsafeGuessKindToResource(gvk.GroupVersion().WithKind(kindName))
+	dm.resourceToKind[singular] = gvk
+	dm.resourceToKind[plural] = gvk
+
+	return dm.updateCachedDynamicRESTMapper()
```

Comment:
make sure no duplicate update cache if GVK has been updated.

Reply:
Thanks for reminding. The recently submitted PR adds validation logic.